server/api.js

```diff
@@ -4,15 +4,140 @@ import pool from "./db";
 
 const router = Router();
 
+// Get all the tasks
 router.get("/tasks", (req, res) => {
 	pool.query("SELECT * FROM tasks").then((result) => {
-		// console.log(result.rows);
 		return res.send(result.rows);
 	});
 });
-
 router.get("/", (_, res) => {
-	return res.json({ message: "Hello, from the api route!" });
+	res.json({ message: "Hello, from the api route!" });
 });
 
+router.post("/tasks", (req, res) => {
+	const { task_title, due_date, evidence, status_title, element_title } =
+		req.body;
+	let params = [];
+	if (!task_title) {
+		return res
+			.status(404)
+			.send({ success: false, message: "Task title cannot be empty" });
+	} else {
+		params.push(task_title);
+	}
+	if (due_date.length > 0) {
+		params.push(due_date);
+	} else {
+		params.push("");
+	}
```

Comment:
This validation is a bit confusing - I think it's saying _"if it's not an empty string add it to the array, otherwise add an empty string"_, how is that different to an unconditional `params.push(due_date)`?

Reply:
Thanks for the review , will work on this .